cli/types/index.d.ts

```diff
@@ -1232,9 +1232,207 @@ declare namespace Cypress {
+    (chainer: 'not.have.thrown', value?: Error | typeof Error | string): Chainable<Subject>
+    (chainer: 'not.returned', value: any): Chainable<Subject>
+
+    // jquery-chai
+    (chainer: 'be.checked'): Chainable<Subject>
+    (chainer: 'be.disabled'): Chainable<Subject>
+    (chainer: 'be.empty'): Chainable<Subject>
+    (chainer: 'be.enabled'): Chainable<Subject>
+    (chainer: 'be.hidden'): Chainable<Subject>
+    (chainer: 'be.selected'): Chainable<Subject>
+    (chainer: 'be.visible'): Chainable<Subject>
+    (chainer: 'contain', value: string): Chainable<Subject>
+    (chainer: 'exist'): Chainable<Subject>
+    (chainer: 'have.attr', value: string, match?: string): Chainable<Subject>
+    (chainer: 'have.class', value: string): Chainable<Subject>
+    (chainer: 'have.css', value: string, match?: string): Chainable<Subject>
+    (chainer: 'have.data', value: string, match?: string): Chainable<Subject>
+    (chainer: 'have.decendants', selector: string): Chainable<Subject>
+    (chainer: 'have.html', value: string): Chainable<Subject>
+    (chainer: 'have.id', value: string, match?: string): Chainable<Subject>
+    (chainer: 'have.prop', value: string, match?: any): Chainable<Subject>
+    (chainer: 'have.text', value: string): Chainable<Subject>
+    (chainer: 'have.value', value: string): Chainable<Subject>
+    (chainer: 'match', value: string): Chainable<Subject>
+
+    // jquery-chai.not
+    (chainer: 'not.be.checked'): Chainable<Subject>
+    (chainer: 'not.be.disabled'): Chainable<Subject>
+    (chainer: 'not.be.empty'): Chainable<Subject>
+    (chainer: 'not.be.enabled'): Chainable<Subject>
+    (chainer: 'not.be.hidden'): Chainable<Subject>
+    (chainer: 'not.be.selected'): Chainable<Subject>
+    (chainer: 'not.be.visible'): Chainable<Subject>
+    (chainer: 'not.contain', value: string): Chainable<Subject>
+    (chainer: 'not.exist'): Chainable<Subject>
+    (chainer: 'not.have.attr', value: string, match?: string): Chainable<Subject>
+    (chainer: 'not.have.class', value: string): Chainable<Subject>
+    (chainer: 'not.have.css', value: string, match?: string): Chainable<Subject>
+    (chainer: 'not.have.data', value: string, match?: string): Chainable<Subject>
+    (chainer: 'not.have.decendants', selector: string): Chainable<Subject>
+    (chainer: 'not.have.html', value: string): Chainable<Subject>
+    (chainer: 'not.have.id', value: string, match?: string): Chainable<Subject>
+    (chainer: 'not.have.prop', value: string, match?: any): Chainable<Subject>
+    (chainer: 'not.have.text', value: string): Chainable<Subject>
+    (chainer: 'not.have.value', value: string): Chainable<Subject>
+    (chainer: 'not.match', value: string): Chainable<Subject>
+
+    // fallback
     (chainers: string, value?: any): Chainable<Subject>
-    (chainers: string, method: string, value: any): Chainable<Subject>
+    (chainers: string, value: any, match: any): Chainable<Subject>

```

Comment:
`value: any` is more accurate here. The first argument isn't always a string for 2-argument assertions.